coderd/coderdtest/oidctest/testidp/main.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+	"os/signal"
+	"testing"
+
+	"github.com/golang-jwt/jwt/v4"
+
+	"github.com/coder/coder/v2/coderd/coderdtest/oidctest"
+)
+
+func main() {
+	testing.Init()
+	_ = flag.Set("test.timeout", "0")
+
+	flag.Parse()
+
+	// This is just a way to run tests outside go test
+	testing.Main(func(pat, str string) (bool, error) {
+		return true, nil
+	}, []testing.InternalTest{
+		{
+			Name: "Run Fake IDP",
+			F:    RunIDP(),
+		},
+	}, nil, nil)
```

Comment:
This feels weird, but I understand you did it to use the `oidctest` package 😄, it's also possible to build a test package, typically I'd suggest to implement this that way, but it wouldn't be as easily discoverable as this is (esp. if put in `cmd/`).

Reply:
I felt werid saying `run go test -run="TestRealIDP"` or something. 

Another idea I had was to just implement the `testing.TB` struct myself and just throw log messages on the Fails(). It's not as trivial as I'd hope though, they have this: https://github.com/golang/go/blob/master/src/testing/testing.go#L899-L902

So I'd have to embed an actual testing.T struct? It just also has it's "jankiness", so just went with this for now.